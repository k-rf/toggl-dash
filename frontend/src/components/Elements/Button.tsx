import { Button as MuiButton, ButtonProps as MuiButtonProps } from "@mui/material";

type Props = Pick<
  MuiButtonProps,
  "children" | "color" | "disabled" | "fullWidth" | "onClick" | "variant"
>;

export const Button = (props: Props) => {
  return <MuiButton {...props} />;
};
