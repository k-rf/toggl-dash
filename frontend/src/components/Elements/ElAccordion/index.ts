export * from "./ElAccordion";
export * from "./ElAccordionDetails";
export * from "./ElAccordionSummary";
