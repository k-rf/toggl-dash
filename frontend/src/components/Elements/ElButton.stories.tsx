import { Box, Stack } from "@mui/material";
import type { Meta, StoryObj } from "@storybook/react";

import { ElButton, ElButtonProps } from "./ElButton";

export default { component: ElButton } as Meta<ElButtonProps>;

type Story = StoryObj<ElButtonProps>;

export const Default: Story = {
  args: {
    children: "Button",
  },
  render: (args) => {
    return <Buttons {...args} />;
  },
};

export const Colors: Story = {
  render: (args) => (
    <Stack direction="column" spacing={2}>
      <Buttons {...args} color="primary">
        Primary
      </Buttons>
      <Buttons {...args} color="secondary">
        Secondary
      </Buttons>
      <Buttons {...args} color="success">
        Success
      </Buttons>
      <Buttons {...args} color="info">
        Info
      </Buttons>
      <Buttons {...args} color="warning">
        Warning
      </Buttons>
      <Buttons {...args} color="error">
        Error
      </Buttons>
      <Buttons {...args} color="inherit">
        Inherit
      </Buttons>
    </Stack>
  ),
};

const Buttons = (args: ElButtonProps) => {
  return (
    <Stack direction="row" spacing={2}>
      <Box width={160}>
        <ElButton {...args} variant="contained" fullWidth />
      </Box>
      <Box width={160}>
        <ElButton {...args} variant="outlined" fullWidth />
      </Box>
      <Box width={160}>
        <ElButton {...args} variant="text" fullWidth />
      </Box>
    </Stack>
  );
};
