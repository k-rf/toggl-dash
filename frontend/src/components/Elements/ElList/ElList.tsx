import { List, ListProps } from "@mui/material";

type Props = Pick<ListProps, "children" | "sx">;

export const ElList = (props: Props) => {
  return <List {...props} />;
};
