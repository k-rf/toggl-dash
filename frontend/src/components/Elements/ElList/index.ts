export * from "./ElList";
export * from "./ElListItem";
