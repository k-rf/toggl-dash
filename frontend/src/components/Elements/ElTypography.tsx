import { Typography, TypographyProps } from "@mui/material";

type Props = Pick<TypographyProps, "children" | "variant" | "color" | "sx">;

export const ElTypography = (props: Props) => {
  return <Typography {...props} />;
};
