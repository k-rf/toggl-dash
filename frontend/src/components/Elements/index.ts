export * from "./ElButton";
export * from "./ElPaper";
export * from "./ElTextField";
export * from "./ElAccordion";
