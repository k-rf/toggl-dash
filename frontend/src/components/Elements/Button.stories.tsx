import { Box, Stack } from "@mui/material";
import type { ComponentMeta, ComponentStoryObj } from "@storybook/react";

import { Button } from "./Button";

type Meta = ComponentMeta<typeof Button>;
type Story = ComponentStoryObj<typeof Button>;

export default { title: "Elements/Button", component: Button } as Meta;

export const Default: Story = {
  args: {
    children: "ボタン",
    color: "primary",
  },
};

export const Variant: Story = {
  ...Default,
  args: {
    ...Default.args,
  },
  render: (args) => {
    return <Buttons {...args} />;
  },
};

export const Disabled: Story = {
  ...Default,
  args: {
    ...Default.args,
    disabled: true,
  },
  render: (args) => {
    return <Buttons {...args} />;
  },
};

export const Colors: Story = {
  ...Default,
  args: {
    ...Default.args,
    fullWidth: true,
  },
  render: (args) => {
    return (
      <Stack direction={"column"} spacing={2}>
        <Buttons {...args} color="primary" width={120}>
          primary
        </Buttons>
        <Buttons {...args} color="secondary" width={120}>
          secondary
        </Buttons>
        <Buttons {...args} color="success" width={120}>
          success
        </Buttons>
        <Buttons {...args} color="warning" width={120}>
          warning
        </Buttons>
        <Buttons {...args} color="error" width={120}>
          error
        </Buttons>
        <Buttons {...args} color="info" width={120}>
          info
        </Buttons>
      </Stack>
    );
  },
};

const Buttons = (args: Story["args"] & { width?: number }) => {
  const { width, ...props } = args;
  return (
    <Stack direction={"row"} spacing={2}>
      <Box width={width}>
        <Button {...props} variant="contained" />
      </Box>
      <Box width={width}>
        <Button {...props} variant="outlined" />
      </Box>
      <Box width={width}>
        <Button {...props} variant="text" />
      </Box>
    </Stack>
  );
};
