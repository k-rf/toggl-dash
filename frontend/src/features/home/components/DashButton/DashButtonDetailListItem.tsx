import { ElListItem } from "~/components/Elements/ElList";
import { ElTypography } from "~/components/Elements/ElTypography";

type Props = {
  label: JSX.Element | string;
  content: JSX.Element | string;
};

export const DashButtonDetailListItem = (props: Props) => {
  return (
    <ElListItem
      sx={{
        p: (theme) => theme.spacing(2, 0, 0),
        flexDirection: "column",
        alignItems: "flex-start",
      }}
    >
      <ElTypography variant="caption">{props.label}</ElTypography>
      {props.content}
    </ElListItem>
  );
};
