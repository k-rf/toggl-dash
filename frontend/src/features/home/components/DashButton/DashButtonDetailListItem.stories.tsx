import type { Meta, StoryObj } from "@storybook/react";

import { DashButtonDetailListItem } from "./DashButtonDetailListItem";

export default { component: DashButtonDetailListItem } as Meta<typeof DashButtonDetailListItem>;

type Story = StoryObj<typeof DashButtonDetailListItem>;

export const Default: Story = {
  args: {
    label: "Detail Label",
    content: "Detail Content",
  },
};
