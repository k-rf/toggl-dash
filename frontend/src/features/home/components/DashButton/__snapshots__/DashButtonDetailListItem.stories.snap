// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features/Home/Components/DashButton/DashButtonDetailListItem > Default 1`] = `
<body>
  <div>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-l7bzlw-MuiListItem-root"
    >
      <span
        class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
      >
        Detail Label
      </span>
      Detail Content
    </li>
  </div>
</body>
`;
